### PR TITLE
Stitching variables invalid json

### PR DESCRIPTION
**Pull request by chris-redhead**
Summary of the changes
Unit test describing issue #2024, and fix to produce valid variables object for stitched requests

N.B. I'm unclear why the `WriteNumber` method was being used, I might be missing something. Looking at the history it was introduced as part of a commit for issue #1277 , but it might have just been a general refactor.

Addresses #2024

### CODE REVIEW

**Comment by PascalSenn** [comment]
related: https://github.com/ChilliCream/hotchocolate/issues/1972

**Comment by chris-redhead** [comment]
> related: #1972

It's essentially the same issue

**Comment by michaelstaib** [comment]
Hi,

thanks for finding that issue and helping with this.

I have reviewed the change and this would actually lead to multiple new issues and allocate more memory. I will have a deeper look at how we should fix that actually tomorrow.

File: src/HotChocolate/Stitching/src/Stitching/Utilities/HttpQueryClient.cs [hunk 1]
```diff
@@ -235,13 +235,11 @@ await httpClient.PostAsync(default(Uri), requestContent)
                         break;
 
                     case IntValueNode i:
-                        jsonWriter.Flush();
-                        WriteNumberValue(writer, i.Value);
+                        jsonWriter.WriteNumberValue(Convert.ToInt32(i.Value));
```
Comment (erwan-joly) [inline src/HotChocolate/Stitching/src/Stitching/Utilities/HttpQueryClient.cs:238]: I think this should be Int64

Comment (michaelstaib) [reply src/HotChocolate/Stitching/src/Stitching/Utilities/HttpQueryClient.cs:238]: The way it is implemented is not good... I think I need to rework this further ... the convert is exactly what I did not want here. Rather better would be just to write the string representation that is already there. Will look at this, this weekend.

Comment (erwan-joly) [reply src/HotChocolate/Stitching/src/Stitching/Utilities/HttpQueryClient.cs:238]: https://github.com/ChilliCream/hotchocolate/pull/2086 is something like that better ?

**Comment by michaelstaib** [comment]
closing this in favor of #2086

**Pull request closed without merging**